### DISCUSSION TITLE
help a young technology adept in lighthouser

### DISCUSSION THREAD

**Discussion by zDeVolTz** [GoogleChrome/lighthouse#15507] [Q&A]
I will forgive you in advance and forgive me if I made a mistake with the choice of topic category and for my English... I recently became acquainted with this wonderful technology and dream of using it to the maximum, but I had doubts that with my unsuccessful approach to the code and its implementation I could have ruined something .

Why I think so: 1. the results of performance tests between desktop and mobile are practically the same, minimizing the code and optimizing it has virtually no effect on performance in my case. Therefore, I’m not sure whether the tests pass in 2 modes in my implementation 2 2. The results of testing directly in the browser using Lighthouse differ from those that I received as an output in the document... 

and here’s a question, if it’s not difficult for you, tell me what I did wrong: improve the analysis.. thanks for any help) 

my improvisation fucntion

`gulp.task('lighthouseR', async () => {
    var array = fs.readFileSync("URLs.csv").toString().split("\n");
    const chrome = await chromeLauncher.launch({ chromeFlags: ["--headless"] })
    
      const mobileOptions = { 
        logLevel: "info",
        output: "csv",
        onlyCategories: ["performance", "accessibility", "best-practices", "seo"],
        audits: [
            "first-meaningful-paint",
            "first-cpu-idle",
            "byte-efficiency/uses-optimized-images",
        ],
        port: chrome.port,
        emulatedFormFactor: 'mobile', 
    };
    
    const desktopOptions = { 
        logLevel: "info",
        output: "csv",
        onlyCategories: ["performance", "accessibility", "best-practices", "seo"],
        audits: [
            "first-meaningful-paint",
            "first-cpu-idle",
            "byte-efficiency/uses-optimized-images",
        ],
        port: chrome.port,
        emulatedFormFactor: 'desktop',
    };
    
    for (let i = 0; i < array.length; i++) {
    
      for (let x = 0; x < 2; x++) {
        if (x == 0) {
            const runnerResult = await lighthouse(array[i], mobileOptions); 
            result.push("URL : " + runnerResult.lhr.finalUrl);
            if (runnerResult.lhr.categories.performance.score) {
                result.push("Mobile Performance : " + runnerResult.lhr.categories.performance.score * 100)
            } else {
                result.push("NA")
            }
        
            if (runnerResult.lhr.categories.accessibility.score) {
                result.push("Mobile Accessibility : " + runnerResult.lhr.categories.accessibility.score * 100)
            } else {
                result.push("NA");
            }
        
            if (runnerResult.lhr.categories["best-practices"].score) {
                result.push("Mobile Best Practices : " + runnerResult.lhr.categories["best-practices"].score * 100)
            } else {
                result.push("NA");
            }
            if (runnerResult.lhr.categories.seo.score) {
                result.push("Mobile SEO : " + runnerResult.lhr.categories.seo.score * 100)
            } else {
                result.push("NA");
            }
        }
        
        else {
            const runnerResultDesktop = await lighthouse(array[i], desktopOptions); 
            if (runnerResultDesktop.lhr.categories.performance.score) {
                result.push("Desktop Performance : " +runnerResultDesktop.lhr.categories.performance.score * 100)
            } else {
                result.push("NA")
            }
        
            if (runnerResultDesktop.lhr.categories.accessibility.score) {
                result.push("Desktop Accessibility : " + runnerResultDesktop.lhr.categories.accessibility.score * 100)
            } else {
                result.push("NA");
            }
        
            if (runnerResultDesktop.lhr.categories["best-practices"].score) {
                result.push("Desktop Best Practices : " + runnerResultDesktop.lhr.categories["best-practices"].score * 100)
            } else {
                result.push("NA");
            }
        
            if (runnerResultDesktop.lhr.categories.seo.score) {
                result.push("Desktop SEO : " + runnerResultDesktop.lhr.categories.seo.score * 100)
            } else {
                result.push("NA");
            }
            }
       
      }
    }

    function convertArrayToCSV(array) {
        let csv = '';
        for (let i = 0; i < array.length; i++) {
            csv += array[i] + '\n';
        }
        return csv;
    }
    
    const csvData = convertArrayToCSV(result);
    
    const bufferData = Buffer.from(csvData, 'utf-8');
    
    fs.writeFileSync('lhreport.csv', bufferData);
    await chrome.kill(); }); 
`

part of my result 

`URL : http://127.0.0.1:5500/src/pages/about.html
Mobile Performance : 99
Mobile Accessibility : 73
Mobile Best Practices : 92
Mobile SEO : 92
Desktop Performance : 99
Desktop Accessibility : 73
Desktop Best Practices : 92
Desktop SEO : 92
URL : http://127.0.0.1:5500/src/pages/contact.html
Mobile Performance : 100
Mobile Accessibility : 77
Mobile Best Practices : 92
Mobile SEO : 92
Desktop Performance : 100
Desktop Accessibility : 77
Desktop Best Practices : 92
Desktop SEO : 92

URL : http://127.0.0.1:5500/src/pages/publication-entry.html
Mobile Performance : 99
Mobile Accessibility : 76
Mobile Best Practices : 92
Mobile SEO : 92
Desktop Performance : 99
Desktop Accessibility : 76
Desktop Best Practices : 92
Desktop SEO : 92 

URL : http://127.0.0.1:5500/build/pages/about.html
Mobile Performance : 100
Mobile Accessibility : 73
Mobile Best Practices : 92
Mobile SEO : 92
Desktop Performance : 100
Desktop Accessibility : 73
Desktop Best Practices : 92
Desktop SEO : 92

URL : http://127.0.0.1:5500/build/pages/contact.html
Mobile Performance : 100
Mobile Accessibility : 77
Mobile Best Practices : 92
Mobile SEO : 92
Desktop Performance : 100
Desktop Accessibility : 77
Desktop Best Practices : 92
Desktop SEO : 92

URL : http://127.0.0.1:5500/build/pages/publication-entry.html
Mobile Performance : 99
Mobile Accessibility : 76
Mobile Best Practices : 92
Mobile SEO : 92
Desktop Performance : 99
Desktop Accessibility : 76
Desktop Best Practices : 92
Desktop SEO : 92`

**Comment by adamraine** [comment]
> the results of performance tests between desktop and mobile are practically the same, minimizing the code and optimizing it has virtually no effect on performance in my case.

I'm not sure exactly what you are asking about. Similar mobile/desktop scores are perfectly normal for web pages. The audit suggesting you minimize code doesn't guarantee any performance improvement, it's just suggesting a best practice that *can* affect performance.

> he results of testing directly in the browser using Lighthouse differ from those that I received as an output in the document

Please read our docs on variability: https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md
